packages/shared_preferences/shared_preferences_android/android/src/main/java/io/flutter/plugins/sharedpreferences/SharedPreferencesPlugin.java

```diff
@@ -5,42 +5,216 @@
 package io.flutter.plugins.sharedpreferences;
 
 import android.content.Context;
+import android.content.SharedPreferences;
+import android.util.Base64;
+import android.util.Log;
 import io.flutter.embedding.engine.plugins.FlutterPlugin;
 import io.flutter.plugin.common.BinaryMessenger;
-import io.flutter.plugin.common.MethodChannel;
+import io.flutter.plugin.common.BinaryMessenger.TaskQueue;
+import io.flutter.plugins.sharedpreferences.Messages.SharedPreferencesApi;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
 
 /** SharedPreferencesPlugin */
-public class SharedPreferencesPlugin implements FlutterPlugin {
-  private static final String CHANNEL_NAME = "plugins.flutter.io/shared_preferences_android";
-  private MethodChannel channel;
-  private MethodCallHandlerImpl handler;
+public class SharedPreferencesPlugin implements FlutterPlugin, SharedPreferencesApi {
```

Comment:
This PR should include filling in `SharedPreferencesTest.java` with tests of the new public API being added here.